### PR TITLE
1.3.0

### PR DESCRIPTION
**Pull request by barrykooij**

### CODE REVIEW

**Comment by Gaya** [comment]
Rejected

**Comment by dannyvankooten** [comment]
CR done 👍 Great work.

**Comment by barrykooij** [comment]
Thanks mi boys

File: assets/js/listings.js [hunk 1]
```diff
@@ -171,10 +185,22 @@ WPCM_Listings.prototype.load_vehicles = function () {
...
+			// set response
+			listings.html( response.listings );
+		}
+
+		// set pagination
+		if ( response.pagination ) {
+			pagination.html( response.pagination );
+
+			instance.bind_pagination();
+		}else {
```
Comment (Gaya) [inline assets/js/listings.js]: `} else {`

File: assets/js/listings.js [hunk 2]
```diff
@@ -171,10 +185,22 @@ WPCM_Listings.prototype.load_vehicles = function () {
 	this.listings.parent().append( jQuery( '<div>' ).addClass( 'wpcm-results-load-overlay' ) );
 	this.listings.parent().append( new WPCM_Spinner().getDOM() );
 
-	jQuery.get( wpcm.ajax_url, args, function ( response ) {
+	jQuery.getJSON( wpcm.ajax_url, args, function ( response ) {
```
Comment (Gaya) [inline assets/js/listings.js]: `getJason`

File: src/SubmitCarHandler.php [hunk 3]
```diff
@@ -284,6 +290,33 @@ public function view_preview() {
...
+
+		// redirect user to login screen if they can't edit this listing
+		if ( ! wp_car_manager()->service( 'user_manager' )->can_edit_listing( $vehicle_id ) ) {
+			wp_redirect( wp_login_url( add_query_arg( 'wpcm_publish', $vehicle_id ) ) );
+			exit;
+		}
+
+		// get vehicle object
+		/** @var Vehicle $vehicle */
+		$vehicle = wp_car_manager()->service( 'vehicle_factory' )->make( $vehicle_id );
```
Comment (dannyvankooten) [inline src/SubmitCarHandler.php]: The following class that extends Pimple may be useful to you.

- https://github.com/ibericode/boxzilla-wp/blob/master/src/di/class-container-with-property-access.php

It allows you to typehint properties on your service container, like this.

- https://github.com/ibericode/boxzilla-wp/blob/master/src/class-boxzilla.php

**Pull request merged**